pkg/tsdb/influxdb/flux/executor.go

```diff
@@ -27,7 +30,21 @@ func executeQuery(ctx context.Context, query queryModel, runner queryRunner, max
 		glog.Warn("Flux query failed", "err", err, "query", flux)
 		dr.Error = err
 	} else {
-		dr = readDataFrames(tables, int(float64(query.MaxDataPoints)*2), maxSeries)
+		// we only enforce a larger number than maxDataPoints
+		maxPointsEnforced := int(float64(query.MaxDataPoints) * maxPointsEnforceFactor)
+		// at the point where the check for exceeding maxPoints happens
+		// we do not have enough information to create a good error-message,
+		// so we create a function here that encapsulates the making
+		// of that error-message
+		makeMaxPointsExceededErrorMessage := func() string {
+			text := fmt.Sprintf("results are truncated, max points tolerance exceeded (max: %d, enforced: %d)", query.MaxDataPoints, maxPointsEnforced)
```

Comment:
Perhaps we can do better in explaining why it's happening:
```
A query returned too many datapoints and the results have been truncated at <enforced> points to prevent memory issues. At the current graph size, Grafana can only draw <maxDatapoints>. 
```

And then add the hint as a sentence if applicable.